File: js/IsUnique.test.js
import { describe, it } from 'mocha';
import { expect } from 'chai';
import { isUnique } from './IsUnique';

describe('isUnique', function () {
  it('should return true for a unique string', function () {
    expect(isUnique('abcde')).to.be.true;
  });
  it('should return false for a non-unique string', function () {
    expect(isUnique('abcdee')).to.be.false;
  });
});

File: java/test/IsUniqueTest.java
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

/**
 * Represents a IsUniqueTest.
 * @author Cam Moore
 *
 */
class IsUniqueTest {

  /**
   * Test method for {@link IsUnique#isUnique(java.lang.String)}.
   */
  @Test
  void testIsUnique() {
    assertTrue("should return true for a unique string", IsUnique.isUnique("abcde"));
    assertFalse("should return false for a non-unique string", IsUnique.isUnique("abcdee"));
  }

  @Test
  public void isUnique() {
    assertTrue("should return true for a unique string", IsUnique.isUnique("abcde"));
    assertFalse("should return false for a non-unique string", IsUnique.isUnique("abcdee"));
  }
}
